src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -64,16 +77,32 @@ public CommandResult execute(Model model) throws CommandException {
                 return new CommandResult(String.format(MESSAGE_SUCCESS, ENTRY_INCOME, GRAPH_CATEGORY), false, false,
                                          true);
             case MONTH:
-                break;
+                // TODO: Add month
+                return null;
             default:
-                break;
+                //should never reach here
+                return null;
             }
-            break;
         default:
             //should never reach here
-            break;
+            return null;
         }
+    }
 
-        throw new CommandException(MESSAGE_INVALID_ENTRY_TYPE);
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof ViewCommand)) {
+            return false;
+        }
+        ViewCommand otherViewCommand = (ViewCommand) other;
+        if (this.month == null) {
+            return this.entryType.equals(otherViewCommand.entryType);
+        }
+        return this.entryType.equals(otherViewCommand.entryType)
```

Comment:
Might need to improve this like by defining getter methods for `month` (return `Optional<T>`), `graphType` and `entryType` like `EditCommand`.

Will add it in a new PR to avoid overloading this